modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/auto/deploy/BatchEngineAutoDeployListenerTest.java

```diff
@@ -485,34 +488,38 @@ private File _toZipFile(String fileName) throws Exception {
 		try (ZipOutputStream zipOutputStream = new ZipOutputStream(
 				new FileOutputStream(zipFile))) {
 
-			Stream<Path> stream = Files.walk(zipFileDirectoryPath);
+			Files.walkFileTree(
+				zipFileDirectoryPath,
+				new SimpleFileVisitor<Path>() {
 
-			stream.forEach(
-				zipEntryPath -> {
-					File zipEntryFile = zipEntryPath.toFile();
+					@Override
+					public FileVisitResult visitFile(
+							Path filePath,
+							BasicFileAttributes basicFileAttributes)
+						throws IOException {
 
-					if (zipEntryFile.isDirectory()) {
```

Comment:
@JanisZhang The `visitFile` method is invoked for both files and directories, so this `if` should be kept.